hudi-common/src/main/java/org/apache/hudi/common/model/HoodieKey.java

```diff
@@ -58,20 +57,18 @@ public boolean equals(Object o) {
       return false;
     }
     HoodieKey otherKey = (HoodieKey) o;
-    return Objects.equal(recordKey, otherKey.recordKey) && Objects.equal(partitionPath, otherKey.partitionPath);
+    return Objects.equals(recordKey, otherKey.recordKey) && Objects.equals(partitionPath, otherKey.partitionPath);
   }
 
   @Override
   public int hashCode() {
-    return Objects.hashCode(recordKey, partitionPath);
+    return Objects.hash(recordKey, partitionPath);
   }
 
   @Override
   public String toString() {
-    final StringBuilder sb = new StringBuilder("HoodieKey {");
```

Comment:
same comment on string builder.. If we print the key in fast path (for debugging), the extra allocations would matter?